client/api/go-client/client.go

```diff
@@ -355,3 +360,30 @@ func (c *ClientOptions) retryDelay(r *http.Response) time.Duration {
 	s := rand.Intn(max-min) + min
 	return time.Second * time.Duration(s)
 }
+
+// CheckRedirectFunc is an alias for the somewhat complex function signature
+// of the CheckRedirect function of the http.Client.
+type CheckRedirectFunc func(*http.Request, []*http.Request) error
+
+// ClientFunc is an alias for the function signature needed to create custom
+// http clients.
+type ClientFunc func(*tls.Config, CheckRedirectFunc) HttpPerformer
+
+// HttpPerformer is an interface that the heketi api client needs from the http
+// client.
+type HttpPerformer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// HeketiHttpClient constructs a new http client for use by the heketi
+// api client, using the traditional heketi approach.
+func HeketiHttpClient(tlsConfig *tls.Config, checkRedirect CheckRedirectFunc) HttpPerformer {
```

Comment:
This would return the default Heketi http client, but I think there is a circular dependency. I can get the default tls config from `DefaultClientOptions()` but `checkRedirect` expects to be a closure captured from an already-created Heketi client.

If my goal is to wrap the default behavior with only an IP address deny list, will redirects be broken? (i.e. token not copied over)